### PR TITLE
Update "Download and deploy ELSER" snippet with adaptive allocations

### PR DESCRIPTION
**Pull request by leemthompo**
Does what it says on tin

cc @linyaru

### CODE REVIEW

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 1]
```diff
@@ -122,16 +122,20 @@ The easiest and recommended way to download and deploy ELSER is to use the {ref}
 ----------------------------------
 PUT _inference/sparse_embedding/my-elser-model
 {
-  "service": "elasticsearch",
+  "service": "elser",
```
Comment (leemthompo) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:125]:
```suggestion
  "service": "elasticsearch",
```

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 2]
```diff
@@ -122,16 +122,20 @@ The easiest and recommended way to download and deploy ELSER is to use the {ref}
...
   "service_settings": {
-    "num_allocations": 1,
-    "num_threads": 1,
-    "model_id": ".elser_model_2_linux-x86_64"
+    "adaptive_allocations": {
+      "enabled": true,
+      "min_number_of_allocations": 3,
+      "max_number_of_allocations": 10
+    },
+    "num_threads": 1
```
Comment (leemthompo) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:132]:
```suggestion
    "num_threads": 1,
    "model_id": ".elser_model_2_linux-x86_64"
```

**Review by leemthompo — COMMENTED** [review]
typos

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 3]
```diff
@@ -124,14 +124,19 @@ PUT _inference/sparse_embedding/my-elser-model
 {
   "service": "elasticsearch",
   "service_settings": {
-    "num_allocations": 1,
+    "adaptive_allocations": {
+      "enabled": true,
+      "min_number_of_allocations": 3,
```
Comment (leemthompo) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:129]:
```suggestion
      "min_number_of_allocations": 1,
```

**Review by kosabogi — APPROVED** [review]

**Pull request merged**